src/ui/pages/AssetInfo/AssetInfo.tsx

```diff
@@ -0,0 +1,253 @@
+    { enabled: ready }
+  );
+
+  const assetAddressPnlQuery = useWalletAssetPnl(
+    {
+      addresses: [params.address],
+      fungibleId: asset_code,
+      currency,
+    },
+    { source: useHttpClientSource() },
+    { enabled: ready }
+  );
+
+  const { data: wallet } = useQuery({
+    queryKey: ['wallet/uiGetCurrentWallet'],
+    queryFn: () => {
+      return walletPort.request('uiGetCurrentWallet');
+    },
+  });
+
+  const chainWithTheBiggestBalance =
+    walletData?.data?.chainsDistribution.at(0)?.chain.id || NetworkId.Zero;
+
+  if (isLoading || !wallet || !walletData) {
+    return (
+      <>
+        <NavigationTitle title={null} documentTitle={`${asset_code} - info`} />
+        <PageColumn style={{ alignItems: 'center', justifyContent: 'center' }}>
+          <CircleSpinner />
+        </PageColumn>
+      </>
+    );
+  }
+
+  invariant(assetFullInfo?.fungible, 'Fungible asset info is missing');
+
+  const isWatchedAddress = isReadonlyAccount(wallet);
+  const isEmptyBalance = walletData?.data.totalValue === 0;
+
+  const chainForSwap = isEmptyBalance
+    ? assetFullInfo.extra.mainChain
+    : chainWithTheBiggestBalance;
+
+  return (
+    <PageColumn>
+      <NavigationTitle
+        title={<AssetPageHeader asset={assetFullInfo.fungible} />}
+        documentTitle={`${assetFullInfo.fungible.name} - info`}
+      />
+      <PageTop />
+      <VStack
+        gap={24}
+        style={{ flexGrow: 1, alignContent: 'start', paddingBottom: 72 }}
+      >
+        <AssetTitleAndChart asset={assetFullInfo.fungible} />
+        <AssetGlobalStats assetFullInfo={assetFullInfo} />
+        <AssetAddressStats
+          address={params.address}
+          wallet={wallet}
+          assetFullInfo={assetFullInfo}
+          walletAssetDetails={walletData.data}
+          assetAddressPnlQuery={assetAddressPnlQuery}
+        />
+        <AssetResources assetFullInfo={assetFullInfo} />
+        <AssetDescription assetFullInfo={assetFullInfo} />
+        <AssetHistory
+          assetId={asset_code}
+          assetFullInfo={assetFullInfo}
+          address={params.address}
+        />
+        <ReportAssetLink asset={assetFullInfo.fungible} />
+      </VStack>
+      {isWatchedAddress || !portfolioData ? null : (
+        <StickyBottomPanel
+          style={{ padding: 0, background: 'none', boxShadow: 'none' }}
+          backdropStyle={{ inset: '-16px -16px 0' }}
+        >
+          <HStack
+            gap={8}
+            style={{
+              width: '100%',
+              gridTemplateColumns: isEmptyBalance ? '1fr' : '1fr auto auto',
+            }}
+          >
+            <Button
+              kind="primary"
+              size={48}
+              as={UnstyledLink}
+              to={
+                isEmptyBalance
+                  ? `/swap-form?chainInput=${chainForSwap}&receiveTokenInput=${asset_code}`
+                  : `/swap-form?chainInput=${chainForSwap}&spendTokenInput=${asset_code}`
```

Comment:
Doesn't SwapForm determine the correct chain itself internally?

Reply:
Let me check.

Reply:
Ok cool!